src/org/etools/j1939_84/bus/j1939/J1939TP.java

```diff
@@ -162,8 +148,14 @@ public void close() {
      * @param stream base stream originally returned from bus.read().
      */
     @Override
-    public Stream<Packet> duplicate(Stream<Packet> stream) {
-        return inbound.duplicate(stream);
+    public Stream<Packet> duplicate(Stream<Packet> stream, int time, TimeUnit unit) {
+        return inbound.duplicate(stream, time, unit);
+    }
+
+    public void error(String msg, Throwable e) {
+        // FIXME where do we want error messages to go?
```

Comment:
Write the errors to the log.

Reply:
"warn" messages and application errors go to the same place? One message is for the manufacturer and the other is for the developers.

Reply:
If there's an error, I think both want to know.  but in this case, the code was just moved so it doesn't need to be resolved right now.